simulator/registry.go

```diff
@@ -60,19 +61,37 @@ type Registry struct {
 	handlers map[types.ManagedObjectReference]RegisterObject
 	locks    map[types.ManagedObjectReference]sync.Locker
 	counter  int
+
+	Namespace string
+	Path      string
 }
 
+var defaultNamespace = strings.TrimPrefix(soap.DefaultVimNamespace, "urn:")
+
 // NewRegistry creates a new instances of Registry
 func NewRegistry() *Registry {
 	r := &Registry{
 		objects:  make(map[types.ManagedObjectReference]mo.Reference),
 		handlers: make(map[types.ManagedObjectReference]RegisterObject),
 		locks:    make(map[types.ManagedObjectReference]sync.Locker),
+
+		Namespace: defaultNamespace,
+		Path:      "/sdk",
 	}
 
 	return r
 }
 
+func (r *Registry) typeFunc(name string) (reflect.Type, bool) {
+	if r.Namespace != "" && r.Namespace != defaultNamespace {
+		kind, ok := defaultMapType(r.Namespace + ":" + name)
```

Comment:
Hi @dougm,

I'm not seeing the issue. I wasn't suggesting that you remove the outer `if` statement. I was just saying you could could reduce the scope of `kind, ok` to a single `if` statement since neither variable is used outside of the `if` boolean decision and inner-most return statement.